src/package-linker.js

```diff
@@ -604,49 +670,61 @@ export default class PackageLinker {
       const ref = pkg._reference;
       invariant(ref, 'Package reference is missing');
 
-      // TODO: We are taking the "shortest" ref tree but there may be multiple ref trees with the same length
-      const refTree = ref.requests.map(req => req.parentNames).sort((arr1, arr2) => arr1.length - arr2.length)[0];
-
-      const getLevelDistance = pkgRef => {
-        let minDistance = Infinity;
-        for (const req of pkgRef.requests) {
-          const distance = refTree.length - req.parentNames.length;
-
-          if (distance >= 0 && distance < minDistance && req.parentNames.every((name, idx) => name === refTree[idx])) {
-            minDistance = distance;
-          }
-        }
+      const refTrees = extractRefTrees(ref);
```

Comment:
Here we build an array of "reference trees" which would be sorted by "name" (the workspace's root reference tree carrying the empty name) and where each entry would represent the corresponding dependency path, instead of only taking the first one which was hanging around.